oauth2_http/java/com/google/auth/oauth2/ImpersonatedCredentials.java

```diff
@@ -724,6 +787,12 @@ public Calendar getCalendar() {
       return this.calendar;
     }
 
+    @Override
+    public Builder setUniverseDomain(String universeDomain) {
```

Comment:
I don't think we have to override this method as it is already public in the parent. I noticed that it seems we are overriding the builder methods in other subclasses of the builder, but I think that was a mistake.

Reply:
I debated this in my mind. One benefit I see overriding the setter allows for a fluent builder API where method calls can be chained without requiring explicit casting to ImpersonatedCredentials.  In this case, I feel it provides a slightly better experience for users setting explicit UD
So user can do 
```
        ImpersonatedCredentials.newBuilder()
            .setUniverseDomain("explicit.domain.com");
```
instead of casting:
```
        (ImpersonatedCredentials) ImpersonatedCredentials.newBuilder()
            .setUniverseDomain("explicit.domain.com");
```
While generally it's best to avoid overriding public methods if possible, override setters in Builders is kind of a special case and more acceptable. .e.g. overriding Builder setters seems to be a pattern in other libraries like Guava ([example](https://github.com/google/guava/blob/d40589de41baa3b2ce8442b24366bc57339b9871/guava/src/com/google/common/collect/ImmutableList.java#L840))

One other concern is removing other overrides in other credential classes will be breaking change to customers code. 
That said, I don't feel too strongly about this, if you have any other concerns, I can remove this. What do you think?

Reply:
I don't think customers have to cast the credential to `ImpersonatedCredentials` to `setUniverseDomain()`, I verified in my local that it works. Can you pleas double check it?
For the Guava example, yes they are doing it because there are some additional logics, but we are not doing anything special other than calling `super.setUniverseDomain()`. So I think we can delete it at this moment and add it later if needed.

Reply:
Confirming that explicit overrides were added to avoid casting.

Reply:
Min and I confirmed that we do need explicit casting right now. However, the problem is that the parent builder's setters should return the child types. I created a [demo PR](https://github.com/googleapis/google-auth-library-java/pull/1536) for it .  Also see how protobuf's [GeneratedMessageV3](https://github.com/protocolbuffers/protobuf/blob/main/java/core/src/main/java/com/google/protobuf/GeneratedMessageV3.java#L158-L159) does it.

That being said, this should be separate problem to solve, so it's OK to override the parent's setters right now.